llvm/lib/Target/RISCV/RISCVFrameLowering.cpp

```diff
@@ -555,9 +555,9 @@ void RISCVFrameLowering::emitPrologue(MachineFunction &MF,
       FirstFrameSetup->getOpcode() == RISCV::CM_PUSH) {
     // Use available stack adjustment in push instruction to allocate additional
     // stack space.
-    uint64_t Spimm = std::min(StackSize, (uint64_t)48);
+    uint64_t Spimm = alignTo(std::min(StackSize, (uint64_t)48), 16);
     FirstFrameSetup->getOperand(1).setImm(Spimm);
-    StackSize -= Spimm;
+    StackSize -= std::min(StackSize, Spimm);
```

Comment:
Does this mean we can move the sp by more than StackSize? Doesn't that make the position of sp out of sync with the rest of the code?

Reply:
Well, if the amount of space in the offset for the `cm.push` is more than the additional stack space required, then yes we could. I don't follow why this is a problem though. Or maybe I am misunderstanding what the `StackSize` variable actually represents - the way I have interpreted it is "The amount of stack space required for parameters and local objects (not including the space to spill CSR's)". Of course, if my interpretation is correct, I'd certainly make the argument that the variable is poorly named, but hey.

Reply:
We can move the sp by more than StackSize here, and if it does, the stack pointer will be readjusted in the following if statement where `StackSize != 0`.